contrib/pax_storage/src/cpp/storage/pax.cc

```diff
@@ -280,17 +282,23 @@ void TableWriter::Open() {
   // insert tuple into the aux table before inserting any tuples.
   cbdb::InsertMicroPartitionPlaceHolder(RelationGetRelid(relation_),
                                         current_blockno_);
+  cur_physical_size_ = 0;
 }
 
 void TableWriter::WriteTuple(TupleTableSlot *slot) {
   Assert(writer_);
   Assert(strategy_);
-  // should check split strategy before write tuple
-  // otherwise, may got a empty file in the disk
-  if (strategy_->ShouldSplit(writer_->PhysicalSize(), num_tuples_)) {
+  // Sampled split check to reduce PhysicalSize() overhead
+  // We first perform a sampled pre-write check to avoid empty files.
+  if ((num_tuples_ % PAX_SPLIT_CHECK_INTERVAL) == 0) {
+    cur_physical_size_ = writer_->PhysicalSize();
+  }
+  
+  if (strategy_->ShouldSplit(cur_physical_size_, num_tuples_)) {
     writer_->Close();
     writer_ = nullptr;
     Open();
+    cur_physical_size_ = 0;
```

Comment:
`cur_physical_size_` has been assigned to 0 in Open() function. The split logic can be in a separate function.